bugs/lp-1456277.txt

```diff
@@ -0,0 +1,319 @@
+=== Begin SRU Template ===
+[Impact]
+We are not allowing for ec2 mirrors to be used on non ec2 instances by default. Users can still
+configure those mirrors manually through user data, but cloud-init will not default to those mirrors
+any longer.
+
+[Test Case]
+
+```
+set -x
+# Manually deploy on a lxc container
+
+cat > setup_dev_proposed.sh <<EOF
+#/bin/bash
+add-apt-repository ppa:cloud-init-dev/proposed -y
+apt-get -qq update
+apt-get -qqy install cloud-init
+EOF
+
+restart_cloudinit() {
+   name=$1
+
+   lxc exec $name -- DI_LOG=stderr /usr/lib/cloud-init/ds-identify --force
+   lxc exec $name -- cloud-init clean --logs
+   lxc exec $name -- cloud-init init --local
+   lxc exec $name -- cloud-init init > /dev/null
+   lxc exec $name -- cloud-init modules --mode=config
+   lxc exec $name -- cloud-init modules --mode=final
+   lxc exec $name -- cloud-init status --wait --long
+}
+
+for SERIES in bionic focal xenial; do
+   echo '=== BEGIN ' $SERIES
+   ref=$SERIES-proposed
+   name=test-$SERIES
+   lxc delete $name --force 2> /dev/null
+   lxc launch ubuntu-daily:$SERIES $name
+
+   echo "###########################################"
+   echo "Configuring metadata"
+   lxc exec $name -- sh -c "echo 'availability-zone: us-east-1b' >> /var/lib/cloud/seed/nocloud-net/meta-data"
```

Comment:
Good approach on this. So that you know, you can specify meta-data at launch time.

```
cat > md <<EOF                                                                  
instance-id: mytest                                                             
local-hostname: mytest                                                          
availability-zone: us-east-1b
public-keys:                                                                    
- <some ssh pubkey to represent from cloud metadata>                            
EOF  

lxc launch ubuntu-daily:$SERIES -c user.meta-data="$(cat md)"